import java.util.*;

class Node {
    int val;
    Node left ;
    Node right;
    Node (int val){
        this.val = val;
    }
}
public class implementation{
    public static void main(String[] args) {
        Node a = new Node (3);//root node 
        Node b = new Node (4);
        Node c = new Node (2);
        
        Node d = new Node (-1);
        Node e = new Node (1);
        Node f = new Node (6);
        Node g = new Node (9);
        a.left = b ; a.right = c;
         b.left = d ; b.right = e;
          c.left = f ; c.right = g;
          display(a);
          levelOrder(a);
    }
    private static void levelOrder(Node root){
        Queue<Node> q = new LinkedList<>();
        q.add(root);
        while(q.size()>0){
            Node front = q.remove();
            System.out.print(front.val+" ");
            if(front.left!=null) q.add(front.left);
            if(front.right!=null) q.add(front.right);
        }
        System.out.println();
    }
